### DISCUSSION TITLE
JetBrains Gateway use latest IDE versions with codespaces

### DISCUSSION THREAD

**Discussion by mazzarito** [community/community#68150] [Codespaces]
### Select Topic Area

General

### Body

Hey All,

Just wanted to share a little trick to use JetBrains Gateway with the default SSH connection and connect to codespaces without having to mess with the codespaces plugin. This allows for utilizing the latest JB Gateway version and will work with pretty much any provider without their stupid buggy plugin... Whether that's the fault of JetBrains API or the provider plugins themselves I'll leave you to decide. I did this on a Mac but the idea should carry over to Linux or Windows with some tweaking. 

You'll need the following:

- Copy of whatever JB Gateway app you want to use
- The latest version of the GH command line utility
- A codespaces workstation to connect to

1. open an SSH connection to codespaces using `gh cs ssh --server-port 57873`
2. Take that port, in my case it was `57873` and use the following settings in jetbrains gateway for the connection:

![image](https://github.com/community/community/assets/1696887/828441ca-4a55-4346-af87-710435fd5cb7)

4. Click `check connection and continue`

6. Choose IDE/version to use 

![image](https://github.com/community/community/assets/1696887/a1eba48c-33e3-4384-91f6-ee8b9cf669df)

**Comment by mazzarito** [comment]
Simplified the instructions a bit- still using this as the codespaces jetbrains plugin seems to never be updated

**Comment by KashMoneyMillionaire** [comment]
To others coming along to this, I had to set my host to something different, the host generated when running `gh cs ssh --config`. My SSH config looked like this:

![image](https://github.com/community/community/assets/5257880/68775a01-0635-4233-8b56-ad0741f94610)

**Comment by nlundquist-lilly** [comment]
For anyone coming to this now, I've had to use "codespace" as the username instead of "vscode". Had to turn to this SSH solution since JetBrains completely pulled the buggy Codespaces plugin from the Gateway plugin marketplace.

**Comment by heathdutton** [comment]
For those still using the SSH workaround: I packaged this into a JetBrains Toolbox plugin that handles the SSH config automatically.

https://github.com/heathdutton/github-codespaces-jetbrains-toolbox

Codespaces appear in Toolbox, and it uses `gh codespace ssh --config` under the hood. Works with latest IDE versions.
